src/domain/modules/modules.repository.ts

```diff
@@ -0,0 +1,32 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { ITransactionApiManager } from '@/domain/interfaces/transaction-api.manager.interface';
+import { IModulesRepository } from '@/domain/modules/modules.repository.interface';
+import { SafesByModule } from '@/domain/modules/entities/safes-by-module.entity';
+
+@Injectable()
+export class ModulesRepository implements IModulesRepository {
+  constructor(
+    @Inject(ITransactionApiManager)
+    private readonly transactionApiManager: ITransactionApiManager,
+  ) {}
+
+  async getSafesByModule(args: {
+    chainId: string;
+    moduleAddress: string;
+  }): Promise<SafesByModule> {
+    const api = await this.transactionApiManager.getTransactionApi(
+      args.chainId,
+    );
+    return await api.getSafesByModule(args.moduleAddress);
```

Comment:
I'd add validation to this function before returning the result to the caller.

I think that we could even use an existing validator, `SafeListValidator`, what do you think?

Reply:
Added in cbc68a8b4ec8699a56ccdd1a95c5226325bec844.